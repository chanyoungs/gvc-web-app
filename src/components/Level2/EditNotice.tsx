import React from "react"

import { INoticeWithMeta } from "../../types"
import { EditNoticePaper } from "../Level1/Papers/EditNoticePaper"

export interface EditNoticeProps {
  notices: INoticeWithMeta[] | undefined
}

export const EditNotice: React.FC<EditNoticeProps> = ({ notices }) => {
  let contents
  if (!notices) {
    contents = <div data-testid="edit-loading">Loading...</div>
  } else if (notices.length === 0) {
    contents = <div data-testid="edit-instructions">광고가 없습니다.</div>
  } else {
    contents = (
      <div>
        {notices.map((notice) => (
          <div key={notice.id} data-testid="edit-notices">
            <EditNoticePaper notice={notice} />
          </div>
        ))}
      </div>
    )
  }
  return contents
}
