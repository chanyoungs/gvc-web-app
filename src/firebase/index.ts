import "firebase/auth"

import * as firebase from "firebase/app"
import "firebase/firestore"

var config = {
  apiKey: "AIzaSyBQf0kjL5VgL9IJuEtqWHevRkOMTBmVwiE",
  authDomain: "london-gvc.firebaseapp.com",
  databaseURL: "https://london-gvc.firebaseio.com",
  projectId: "london-gvc",
  storageBucket: "london-gvc.appspot.com",
  messagingSenderId: "139635867699",
  appId: "1:139635867699:web:649f0b856111fee7"
}

export const app = firebase.initializeApp(config)
export const auth = firebase.auth
export const db = firebase.firestore()