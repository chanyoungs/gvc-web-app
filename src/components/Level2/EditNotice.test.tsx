import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import EnzymeAdapter from 'enzyme-adapter-react-16'
import { EditNotice } from './EditNotice'

import firebase from '../../firebase/index'

Enzyme.configure({
  adapter: new EnzymeAdapter(),
  disableLifecycleMethods: true
})

describe('<EditNotice /> with undefined', () => {
  const container = shallow(<EditNotice notices={undefined} />)

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('renders without error', () => {
    expect(
      container.find('[data-testid="component-edit-notice"]').length
    ).toEqual(1)
  })

  it('renders instructions to edit without error', () => {
    expect(container.find('[data-testid="edit-loading"]').length).toEqual(1)
  })
})

describe('<EditNotice /> with no props', () => {
  const container = shallow(<EditNotice notices={[]} />)

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('renders without error', () => {
    expect(
      container.find('[data-testid="component-edit-notice"]').length
    ).toEqual(1)
  })

  it('renders instructions to edit without error', () => {
    expect(container.find('[data-testid="edit-instructions"]').length).toEqual(
      1
    )
  })
})

describe('<EditNotice /> with other props', () => {
  const initialProps = {
    notices: [
      {
        content: 'content1',
        createdAt: firebase.firestore.Timestamp.now(),
        id: 'id1',
        title: 'title1'
      },
      {
        content: 'content2',
        createdAt: firebase.firestore.Timestamp.now(),
        id: 'id2',
        title: 'title2'
      },
      {
        content: 'content3',
        createdAt: firebase.firestore.Timestamp.now(),
        id: 'id3',
        title: 'title3'
      }
    ]
  }
  const container = shallow(<EditNotice {...initialProps} />)

  it('should match the snapshot', () => {
    expect(container.html()).toMatchSnapshot()
  })

  it('renders without error', () => {
    expect(
      container.find('[data-testid="component-edit-notice"]').length
    ).toEqual(1)
  })

  it('renders instructions to edit without error', () => {
    expect(container.find('[data-testid="edit-notices"]').length).toEqual(
      initialProps.notices.length
    )
  })
})
