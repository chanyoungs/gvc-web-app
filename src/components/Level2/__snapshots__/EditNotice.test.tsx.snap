// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<EditNotice /> with no props should match the snapshot 1`] = `"<div data-testid=\\"edit-instructions\\">광고가 없습니다.</div>"`;

exports[`<EditNotice /> with other props should match the snapshot 1`] = `"<div><div data-testid=\\"edit-notices\\"><div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div data-testid=\\"component-edit-noticepaper\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"title\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Title</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" value=\\"title1\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"content\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"standard-multiline-flexible\\" id=\\"standard-multiline-flexible-label\\">Content</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline\\"><textarea rows=\\"1\\" aria-invalid=\\"false\\" id=\\"standard-multiline-flexible\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\">content1</textarea><textarea aria-hidden=\\"true\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility:hidden;position:absolute;overflow:hidden;height:0;top:0;left:0;transform:translateZ(0)\\"></textarea></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit\\"><span class=\\"MuiButton-label\\">Update notice</span></button></div></div></div><div data-testid=\\"edit-notices\\"><div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div data-testid=\\"component-edit-noticepaper\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"title\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Title</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" value=\\"title2\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"content\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"standard-multiline-flexible\\" id=\\"standard-multiline-flexible-label\\">Content</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline\\"><textarea rows=\\"1\\" aria-invalid=\\"false\\" id=\\"standard-multiline-flexible\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\">content2</textarea><textarea aria-hidden=\\"true\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility:hidden;position:absolute;overflow:hidden;height:0;top:0;left:0;transform:translateZ(0)\\"></textarea></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit\\"><span class=\\"MuiButton-label\\">Update notice</span></button></div></div></div><div data-testid=\\"edit-notices\\"><div class=\\"MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded\\"><div data-testid=\\"component-edit-noticepaper\\"><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"title\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\">Title</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl\\"><input type=\\"text\\" aria-invalid=\\"false\\" value=\\"title3\\" class=\\"MuiInputBase-input MuiInput-input\\"/></div></div><div class=\\"MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth\\" data-testid=\\"content\\"><label class=\\"MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiFormLabel-filled\\" data-shrink=\\"true\\" for=\\"standard-multiline-flexible\\" id=\\"standard-multiline-flexible-label\\">Content</label><div class=\\"MuiInputBase-root MuiInput-root MuiInput-underline MuiInputBase-fullWidth MuiInput-fullWidth MuiInputBase-formControl MuiInput-formControl MuiInputBase-multiline MuiInput-multiline\\"><textarea rows=\\"1\\" aria-invalid=\\"false\\" id=\\"standard-multiline-flexible\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\">content3</textarea><textarea aria-hidden=\\"true\\" class=\\"MuiInputBase-input MuiInput-input MuiInputBase-inputMultiline MuiInput-inputMultiline\\" readonly=\\"\\" tabindex=\\"-1\\" style=\\"visibility:hidden;position:absolute;overflow:hidden;height:0;top:0;left:0;transform:translateZ(0)\\"></textarea></div></div><button class=\\"MuiButtonBase-root MuiButton-root MuiButton-text\\" tabindex=\\"0\\" type=\\"button\\" data-testid=\\"submit\\"><span class=\\"MuiButton-label\\">Update notice</span></button></div></div></div></div>"`;

exports[`<EditNotice /> with undefined should match the snapshot 1`] = `"<div data-testid=\\"edit-loading\\">Loading...</div>"`;
